lib/crypto.js

```diff
@@ -0,0 +1,166 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+
+import {
+  sign,
+  verify,
+  createPrivateKey,
+  createPublicKey
+} from 'crypto';
+
+// used to export node's public keys to buffers
+const publicKeyEncoding = {format: 'der', type: 'spki'};
+// used to export node's private keys to buffers
+const privateKeyEncoding = {format: 'der', type: 'pkcs8'};
+// used to turn private key bytes into a buffer in DER format
+const DER_PRIVATE_KEY_PREFIX = Buffer.from(
+  '302e020100300506032b657004220420', 'hex');
+// used to turn public key bytes into a buffer in DER format
+const DER_PUBLIC_KEY_PREFIX = Buffer.from('302a300506032b6570032100', 'hex');
+
+/**
+ * The key material is the part of the buffer after the DER Prefix.
+ *
+ * @param {Buffer} buffer - A DER encoded key buffer.
+ *
+ * @throws {Error} If the buffer does not contain a valid DER Prefix.
+ *
+ * @returns {Buffer} The key material part of the Buffer.
+*/
+function getKeyMaterial(buffer) {
+  if(buffer.indexOf(DER_PUBLIC_KEY_PREFIX, 0) === 0) {
+    return buffer.slice(DER_PUBLIC_KEY_PREFIX.length, buffer.length);
+  }
+  if(buffer.indexOf(DER_PRIVATE_KEY_PREFIX, 0) === 0) {
```

Comment:
```suggestion
  if(buffer.indexOf(DER_PRIVATE_KEY_PREFIX) === 0) {
```

Reply:
same as above, I think it's better to specify 0 here so that if some combination of the prefix and the private key material happens to match the prefix we don't end up matching that.

Reply:
I think there's confusion over what [the API](https://nodejs.org/api/buffer.html#buffer_buf_indexof_value_byteoffset_encoding) does. Passing `0` or not passing `0` does the same thing, as `0` is the default value for the second parameter. It is the check `=== 0` that ensures that the prefix is at the start of the buffer.